pulsar-client/src/main/java/org/apache/pulsar/client/impl/MessageCrypto.java

```diff
@@ -118,7 +120,7 @@
 
     public MessageCrypto(boolean keyGenNeeded) {
 
-        encryptedDataKeyMap = new ConcurrentHashMap<String, byte[]>();
+        encryptedDataKeyMap = new ConcurrentHashMap<String, Pair<byte[], List<KeyValue>>>();
```

Comment:
if we already have `EncryptionKeyInfo` which contains `byte[]` and `List<KeyValue>>` then can we use the same instead `Pair`?

Reply:
Fixed. The one difference here is that this map stores encrypted key vs the KeyReader returns actual key.